File: src/main/java/com/example/demo/BookController.java
package com.example.demo;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping(produces = MediaType.APPLICATION_JSON_VALUE)
public class BookController {

    private final BookRepository repository;

    public BookController(BookRepository repository) {
        this.repository = repository;
    }

    @GetMapping(path = "/books")
    public List<Book> getAllBooks() {
        return this.repository.findAll();
    }

    @GetMapping(path = "/books/{title}")
    public Book byTitle(@PathVariable String title) {
        return this.repository.findByTitleIgnoringCase(title);
    }

}

File: src/main/kotlin/com/example/demo/BookController.kt
package com.example.demo

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController

@RestController
class BookController(private val repository: BookRepository) {

    @GetMapping("/books")
    fun allBooks() = repository.findAll()

    @GetMapping("/books/{title}")
    fun byTitle(@PathVariable title: String) = repository.findByTitleIgnoringCase(title)

}
